students/brandon_nguyen/session03/mailroom.py

```diff
@@ -0,0 +1,114 @@
+import sys
+import unittest
+from decimal import Decimal
+
+#!/usr/bin/env python3
+#Week3 Excercise mailroom part 1
+#Student: Brandon Nguyen - Au2018
+
+#global define data structure
+donor_db = [("Brandon Nguyen", [4500.33,350.87,300.05]),
+          ("Shawn DeGraw", [5500.00, 345.05,571.33]),
+          ("Jacqueline Lee",[4300.00,3200.00,230.13]),
+          ("Aidan Nguyen", [4300.00,200.00,238.23]),
+          ]
+promptTxt = "\n".join(("Welcome to the mailroom program!",
+          "Please choose from below 3 options:",
+          "1 - Send a Thank You",
+          "2 - Create a Report",
+          "3 - Quit",
+          ">>> "))
+#The script should prompt the user (you) to choose from a menu of 3 actions: “Send a Thank You”, “Create a Report” or “quit”.
+
+#think about a main menu function that can be called any where.
+def main_menu():
+    promptTxt = "\n".join(("Welcome to the mailroom program!",
+          "Please choose from below 3 options:",
+          "1 - Send a Thank You",
+          "2 - Create a Report",
+          "3 - Quit",
+          ">>> "))
+
+def send_ty():
+    while True:
+        response = input("\nPlease enter donor full names or \"list\". Q to return to main menu: " )
+        if response.strip() == 'list':
+            list_donors()
+            break
+        elif response.strip() == 'Q':
+            main()
+        else:
+            for name in donor_db:
+            #print(name[0])
+                if name[0] == response.strip():
+                    indx = donor_db.index(name)
+                    while True:
+                        donation = input("Please enter donation amount for " + name[0] +". Q to quit. :>>")
+                        if donation.strip() == 'Q':
+                            main()
```

Comment:
Again, not a good use of recursion. Since you are already in another nested loop, `break` may not work. However, you are in a function, and you just need to get out of the function in this case, so a simple `return` here will just work.